app/screens/channel_add_people/channel_add_people.tsx

```diff
@@ -0,0 +1,340 @@
+    const [searchResults, setSearchResults] = useState<UserProfile[]>([]);
+    const [loading, setLoading] = useState(false);
+    const [term, setTerm] = useState('');
+    const [startingConversation, setStartingConversation] = useState(false);
+    const [selectedIds, setSelectedIds] = useState<{[id: string]: UserProfile}>({});
+    const selectedCount = Object.keys(selectedIds).length;
+    const groupConstrained = currentChannel.isGroupConstrained;
+    const currentChannelId = currentChannel.id;
+
+    const isSearch = Boolean(term);
+
+    const loadedProfiles = ({users}: {users?: UserProfile[]}) => {
+        if (mounted.current) {
+            if (users && !users.length) {
+                next.current = false;
+            }
+
+            page.current += 1;
+            setLoading(false);
+            dispatchProfiles({type: 'add', values: users});
+        }
+    };
+
+    const getProfiles = useCallback(debounce(() => {
+        if (next.current && !loading && !term && mounted.current) {
+            setLoading(true);
+            fetchProfilesNotInChannel(serverUrl,
+                currentTeamId,
+                currentChannelId,
+                groupConstrained,
+                page.current + 1,
+                General.PROFILE_CHUNK_SIZE).then(loadedProfiles);
+        }
+    }, 100), [loading, isSearch, serverUrl, currentTeamId]);
+
+    const handleRemoveProfile = useCallback((id: string) => {
+        const newSelectedIds = Object.assign({}, selectedIds);
+
+        Reflect.deleteProperty(newSelectedIds, id);
+
+        setSelectedIds(newSelectedIds);
+    }, [selectedIds]);
+
+    const addPeopleToChannel = useCallback(async (ids: string[]): Promise<boolean> => {
+        const result = await addMembersToChannel(serverUrl, currentChannelId, ids, '', false);
+
+        if (result.error) {
+            alertErrorWithFallback(intl, result.error, messages.error);
+        }
+
+        return !result.error;
+    }, [serverUrl]);
+
+    const clearSearch = useCallback(() => {
+        setTerm('');
+        setSearchResults([]);
+    }, []);
+
+    const startConversation = useCallback(async (selectedId?: {[id: string]: boolean}) => {
+        if (startingConversation) {
+            return;
+        }
+
+        setStartingConversation(true);
+
+        const idsToUse = selectedId ? Object.keys(selectedId) : Object.keys(selectedIds);
+        let success;
+        if (idsToUse.length === 0) {
+            success = false;
+        } else {
+            success = await addPeopleToChannel(idsToUse);
+        }
+
+        if (success) {
+            close();
+        } else {
+            setStartingConversation(false);
+        }
+    }, [startingConversation, selectedIds, addPeopleToChannel]);
+
+    const handleSelectProfile = useCallback((user: UserProfile) => {
+        if (selectedIds[user.id]) {
+            handleRemoveProfile(user.id);
+            return;
+        }
+
+        if (user.id === currentUserId) {
```

Comment:
Is this even possible? Can I be in this screen while I am not a member? Should I not be in the list of members and therefore this part of the code never be executed?

Reply:
You're correct. In this case, the current user is a member of the channel and doesn't show up in the list of users.